matchmaking-ext/end_participation.go

```diff
@@ -11,7 +11,35 @@ import (
 
 func endParticipation(err error, client *nex.Client, callID uint32, idGathering uint32, strMessage string) {
 	server := commonMatchMakingExtProtocol.server
-	matchmakeSession := common_globals.Sessions[idGathering].GameMatchmakeSession
+	session, ok := common_globals.Sessions[idGathering]
+	if !ok {
```

Comment:
The more idiomatic way to write this in Go would be

```suggestion
	if session, ok := common_globals.Sessions[idGathering]; !ok {
```

Reply:
This doesn't appear to assign the session var outside of the if statement. I could put the rest in an else{}, that should work, up to you

Reply:
You have to set the following right before if statement:

```go
var session *common_globals.CommonMatchmakeSession
```

Reply:
Yes that's true, when done this way the variable is scoped to the following block. I would suggest then defining the variables inside the scope they are supposed to be used in, then. Like so

```suggestion
	var session *common_globals.CommonMatchmakeSession
	var ok bool
	if session, ok = common_globals.Sessions[idGathering]; !ok {
```